File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.ts
function minimumDeletions(s: string): number {
    let bCount = 0;
    let deletion = 0;
    for(let i = 0; i < s.length; i++) {
        if(s[i] === 'b') bCount++
        // delete 'a' or previous b
        else deletion = Math.min(deletion + 1, bCount)
    }
    return deletion;
};

File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.go
func minimumDeletions(s string) int {
    deletion := 0;
    bCount := 0;
    for i := 0; i < len(s); i++ {
        if s[i] == 'b' {
            bCount++
        } else {
            deletion = min(deletion + 1, bCount)
        }
    }
    return deletion;
}